ceph_deploy/util/init_systems.py

```diff
@@ -0,0 +1,246 @@
+            )
+        cmd.append('{action}'.format(action=action))
+        if init_script == 'ceph':
+            cmd.extend(
+                [
+                    '{service}.{name}'.format(
+                        service=SYSV_SERVICES.get(service, service),
+                        name=daemon_name
+                    )
+                ]
+            )
+        return cmd
+
+    def start(self, service, **kw):
+        cluster = kw.pop('cluster', 'ceph')
+        name = kw.pop('name', self.hostname)
+        cmd = self._build_service_command(
+                service, 'start', name, cluster_name=cluster
+        )
+        self._run(cmd)
+
+    def stop(self, service, **kw):
+        name = kw.pop('name', self.hostname)
+        cmd = self._build_service_command(service, 'stop', name)
+        self._run(cmd)
+
+    def enable(self, service, **kw):
+        init_script = self._get_init_script(service)
+        self._run(
+            [
+                'chkconfig',
+                init_script,
+                'on'
+            ]
+        )
+
+    def disable(self, service, **kw):
+        init_script = self._get_init_script(service)
+        self._run(
+            [
+                'chkconfig',
+                init_script,
+                'off'
+            ]
+        )
+
+    def status(self, service, **kw):
+        name = kw.pop('name', self.hostname)
+        cmd = self._build_service_command(service, 'status', name)
+        self._run(cmd)
+
+
+class Upstart(InitSystem):
+    """
+    Upstart Init System
+    """
+
+    name = 'upstart'
+
+    def start(self, service, **kw):
+        cluster = kw.pop('cluster', 'ceph')
+        name = kw.pop('name', self.hostname)
+        self._run(
+            [
+                'initctl',
+                'emit',
+                '{service}'.format(
+                    service=UPSTART_SERVICES.get(service, service)
+                ),
+                'cluster={cluster}'.format(cluster=cluster),
+                'id={name}'.format(name=name)
+            ]
+        )
+
+    def stop(self, service, **kw):
+        name = kw.pop('name', self.hostname)
+        self._run(
+            [
+                'stop',
+                '{service}'.format(
+                    service=UPSTART_SERVICES.get(service, service)
+                ),
+                'id={name}'.format(name=name)
+            ]
+        )
+
+    def enable(self, service, **kw):
+        pass
+
+    def disable(self, service, **kw):
+        pass
+
+    def status(self, service, **kw):
+        cluster = kw.pop('cluster', 'ceph')
+        name = kw.pop('name', self.hostname)
+        self._run(
+            [
+                'initctl',
+                'status',
+                '{service}'.format(
+                    service=UPSTART_SERVICES.get(service, service)
+                ),
+                'cluster={cluster}'.format(cluster=cluster),
+                'id={name}'.format(name=name)
+            ]
+        )
+
```

Comment:
I don't think any of these need return values.  the `_run()` method is calling `remoto.process.run()`, which will raise an exception (by default) if the the command returns non-zero.

If any of our commands on remote nodes return non-zero, we have ceph-deploy halt it's execution.  I'm not looking to change behavior here.

That being said, this does highlight that I call `return` in `InitSystem._run()`, which doesn't make any sense.  I can remove that.